llvm/lib/Target/DirectX/DXILRootSignature.cpp

```diff
@@ -186,20 +187,23 @@ PreservedAnalyses RootSignatureAnalysisPrinter::run(Module &M,
   OS << "Root Signature Definitions"
      << "\n";
   uint8_t Space = 0;
-  for (const auto &P : RSDMap) {
-    const auto &[Function, RSD] = P;
-    OS << "Definition for '" << Function->getName() << "':\n";
+  for (const auto &F : M) {
```

Comment:
Better style to spell out the type if it doesn't show up elsewhere (such as in a cast).
```suggestion
  for (const Function &F : M) {
```